### PR TITLE
libraries all unit-tested

### PR DESCRIPTION
**Pull request by itsHaseebSaeed**
Completed unit-test of all libraries. relatively small PR but helps understand few changes made in core functionality of libraries.

### CODE REVIEW

**Comment by itsHaseebSaeed** [comment]
@DrPresident Please review this PR, Just added lb-token(Snip-1155)

File: contracts/liquidity_book/lb_pair/src/contract.rs [hunk 1]
```diff
@@ -72,21 +74,22 @@ pub fn instantiate(
...
+        entropy: msg.entropy,
+        lb_pair_info: LbPair {
+            name: format!(
+                "{}-{}-lbtoken-{}",
+                &msg.token_x.unique_key(),
+                &msg.token_y.unique_key(),
+                &msg.bin_step
+            ),
+            symbol: format!("{}/{}-LB-{} LP", "tokenX", "tokenY", &msg.bin_step),
+            lb_pair_address: env.contract.address.clone(),
```
Comment (DrPresident) [inline contracts/liquidity_book/lb_pair/src/contract.rs:89]: Why does the token care what the pair address is?

File: contracts/liquidity_book/lb_pair/src/contract.rs [hunk 2]
```diff
@@ -72,21 +74,22 @@ pub fn instantiate(
...
-        lb_pair: env.contract.address.clone(),
+        has_admin: false,
+        admin: None,
+        curators: [env.contract.address.clone()].to_vec(),
+        entropy: msg.entropy,
+        lb_pair_info: LbPair {
+            name: format!(
+                "{}-{}-lbtoken-{}",
+                &msg.token_x.unique_key(),
+                &msg.token_y.unique_key(),
```
Comment (DrPresident) [inline contracts/liquidity_book/lb_pair/src/contract.rs:85]: what do these unique keys look like?

File: contracts/liquidity_book/lb_pair/src/contract.rs [hunk 3]
```diff
@@ -72,21 +74,22 @@ pub fn instantiate(
...
+                &msg.bin_step
+            ),
+            symbol: format!("{}/{}-LB-{} LP", "tokenX", "tokenY", &msg.bin_step),
```
Comment (DrPresident) [inline contracts/liquidity_book/lb_pair/src/contract.rs:88]: This looks to be set to `tokenX/tokenY-LB-1`

**Review by DrPresident — APPROVED** [review]
None of the tests in `lb_pair` pass, and the `lb_token` won't run any tests with `error: current package believes it's in a workspace when it's not`

Not worried about these issues at this stage so I'll approve

File: contracts/liquidity_book/lb_pair/src/contract.rs (hunk 1 above)
Comment (itsHaseebSaeed) [reply contracts/liquidity_book/lb_pair/src/contract.rs:89]: I just removed that in the upcoming PR.

File: contracts/liquidity_book/lb_pair/src/contract.rs (hunk 2 above)
Comment (itsHaseebSaeed) [reply contracts/liquidity_book/lb_pair/src/contract.rs:85]: Just the address

**Comment by itsHaseebSaeed** [comment]
> None of the tests in `lb_pair` pass, and the `lb_token` won't run any tests with `error: current package believes it's in a workspace when it's not`
> 
> Not worried about these issues at this stage so I'll approve

well just added new tests, cause lb-pair tests won't work without multi-tests. lb-token is non functional as of this PR. Opening a big PR soon.

**Pull request merged**